### DISCUSSION TITLE
Use Nx release to version apps

### DISCUSSION THREAD

**Discussion by ak274** [nrwl/nx#29445] [Questions]
Hi @JamesHenry!
 
We have multiple apps in our workspace which are dependent on many non-buildable and buildable libs.. Now i want to understand how we can configure the release in `nx.json ` to support versioning and change logs? If we make a change in a lib which is dependency to one app it should only version that app and not all the apps in the `/apps`  folder.

Currently my config is following but this config always do version for all the apps in the folder instead of it i want it to do the versioning to only affected apps :

```
"release": {
    "projectsRelationship": "independent",
    "releaseTagPattern": "{projectName}-{version}",
    "projects": ["apps/*", "!apps/docs"],
    "git": {
      "commitMessage": "chore({projectName}): release version ${version} [skip ci]"
    },
    "changelog": {
      "projectChangelogs": true
    }
  }

```

**Comment by thdk** [comment]
Can you provide some more about your work flow?

- which files did you change since the previous release?
   - only files in the lib?
   - or also files in the root directory of your repo?
- how do you initiate a release?
   - `nx release`
       - it should prompt you for a specifier for each project
          - so all your apps will appear here, but you only have to select the single app

**Reply by ak274** [reply]
For us the only thing which matters in terms of release are apps; libs are non-buildable and non-publishable. They are dependencies of our apps. And if any changes to a dependent made it should run version and changelog for that app which is impacted if multiple apps were impacted then for all its should run those commands.   

So for this question:

which files did you change since the previous release?
only files in the lib? (if only this happen it should do release of that particular app which is impacted)
or also files in the root directory of your repo? (if this happens we can release each app which is dependent )

how do you initiate a release? We do this in our gitlab CICD so projects which are release candidates  should be affected only apps
nx release
it should prompt you for a specifier for each project
so all your apps will appear here, but you only have to select the single app
